tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -759,9 +793,29 @@ private async Task<ObjectPackFilesResult> GetObjectsPackFileFromWorkingDirectory
             return new ObjectPackFilesResult(Array.Empty<string>(), temporaryFolder);
         }
 
-        Log.Debug("ITR: Packing objects...");
+        // Sanitize object list (on some cases we get a "fatal: expected object ID, got garbage" error because the object list has invalid escape chars)
+        var objectsOutput = getObjectsCommand!.Output;
+        var lstObjectsSha = new List<string>();
+        foreach (Match? match in ShaRegex.Matches(objectsOutput))
```

Comment:
You expect most of these matches to be valid right? If so you could preallocate list?
```suggestion
        var matches = ShaRegex.Matches(objectsOutput);
        var lstObjectsSha = new List<string>(matches.Count);
        foreach (Match? match in matches)
```

Reply:
True, done in [100c905](https://github.com/DataDog/dd-trace-dotnet/pull/5232/commits/100c905e66821a8e3b69a08e6732f2af760b5a8a)